deepchecks/nlp/checks/data_integrity/frequent_substrings.py

```diff
@@ -0,0 +1,227 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module contains Frequent Substrings check."""
+import typing as t
+from collections import defaultdict
+from typing import Dict
+
+import pandas as pd
+
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.nlp import Context, SingleDatasetCheck
+from deepchecks.nlp._shared_docs import docstrings
+from deepchecks.nlp.text_data import TextData
+from deepchecks.utils.strings import format_percent
+
+__all__ = ['FrequentSubstrings']
+
+
+@docstrings
+class FrequentSubstrings(SingleDatasetCheck):
+    """Checks for frequent substrings in the dataset.
+
+    Parameters
+    ----------
+    n_to_show : int, default: 5
+        number of most common duplicated samples to show.
+    n_samples : int, default: 10_000_000
+        number of samples to use for this check.
+    random_state : int, default: 42
+        random seed for all check internals.
+    min_threshold: float, default: 0.05
+        minimum threshold for tagging a substring as frequent
+    min_ngram: int, default: 4
+        minimum n for the n-gram extraction
```

Comment:
Elaborate a bit more here. E.g. "Minimum amount of words for a substring to be considered a frequent substring"